src/renderer/tvgAnimation.cpp

```diff
@@ -93,6 +94,35 @@ float Animation::duration() const noexcept
 }
 
 
+Result Animation::segment(float begin, float end) noexcept
+{
+    if (begin < 0.0 || end > 1.0 || begin >= end) return Result::InvalidArguments;
+
+    auto loader = pImpl->picture->pImpl->loader;
+    if (!loader) return Result::InsufficientCondition;
+    if (!loader->animatable()) return Result::NonSupport;
+
+    static_cast<LottieLoader*>(loader)->segment(begin, end);
+
+    return Result::Success;
+}
+
+
+Result Animation::segment(float *begin, float *end) noexcept
+{
+    auto loader = pImpl->picture->pImpl->loader;
+    if (!loader) return Result::InsufficientCondition;
+    if (!loader->animatable()) return Result::NonSupport;
+    if (!begin || !end) return Result::InvalidArguments;
```

Comment:
I think API could can return either one.
```
if (!begin && !end) return Result::InvalidArguments;
```

Reply:
Updated API for the case returns in single param.
```cpp
animation->segment(&begin);
animation->segment(nullptr, &end)
```